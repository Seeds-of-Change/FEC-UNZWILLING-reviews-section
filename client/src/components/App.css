@font-face {
  font-family: 'futura_bkbook';
  src: url('/fonts/futura_book-webfont.woff2') format('woff2'),
       url('/fonts/futura_book-webfont.woff') format('woff');
  font-weight: normal;
  font-style: normal;
}

.big_grey_container {
  margin: -8px; /*def a better way to do this*/
  padding: 10%;
  background-color: rgb(242,240,238);
}

.main_white_container {
  background-color: rgb(255,255,255);
}

.white_container_with_text {
  padding-right: 10%;
  padding-left: 10%;
  padding-top: 8%;
  padding-bottom: 5%;

  word-wrap: break-word;
  line-height: 1.5em;
  font-family: Futura,Trebuchet MS,Arial,sans-serif;
  font-weight: lighter;
  /* font-family: "futura book"!; */
  font-size: 14px;
  line-height: 20px;
}

.reviews_questions_block {
  display: flex;
}

.horizontal_line {
  border-top: 1px solid black;
  border-bottom: 1px solid black;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-left: 20px;
  width: 100%
}

/* margin is the space around an element and padding refers to the space between an element and the content inside it. */
.reviewHeading {
  margin-right: 150px;
}