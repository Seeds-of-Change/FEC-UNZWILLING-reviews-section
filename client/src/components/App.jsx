/* eslint-disable no-console */
import React from 'react';
import axios from 'axios';

import IndividualReview from './ReviewContent/IndividualReview';

import ReviewOverview from './SectionSummary/ReviewOverview';
import MostHelpfulReviews from './SectionSummary/MostHelpfulReviews';

import PaginationAndSort from './Pagination/PaginationAndSort';
import PaginationAndNextPage from './Pagination/PaginationAndNextPage';
/*--------------------------------*/
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      oneItem: {},
    };
  }

  componentDidMount() {
    this.getReviews();
  }

  getReviews() {
    axios.get('http://localhost:3000/review/1')
      .then((response) => {
        this.setState({
          oneItem: response.data,
        });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    const {
      aggregate_five_star_review,
      aggregate_four_star_review,
      aggregate_three_star_review,
      aggregate_two_star_review,
      aggregate_one_star_review,
      aggregate_value_rating,
      aggregate_quality_rating,
      aggregate_star_rating,
      total_number_reviews,
      reviews,
      most_helpful_critical,
      most_helpful_favorable,
    } = this.state.oneItem;
    return (
      <div className="container">
        <div className="block">
          <span className="hover-hand">Reviews&nbsp;&nbsp;</span>
          <span className="hover-hand">Questions</span>
        </div>
        <div>
          <ReviewOverview
            numberOfFiveStarReviews={aggregate_five_star_review}
            numberOfFourStarReviews={aggregate_four_star_review}
            numberOfThreeStarReviews={aggregate_three_star_review}
            numberOfTwoStarReviews={aggregate_two_star_review}
            numberOfOneStarReviews={aggregate_one_star_review}
            averageValueRating={aggregate_value_rating}
            averageQualityRating={aggregate_quality_rating}
            averageStarRating={aggregate_star_rating}
          />
          <MostHelpfulReviews
            reviewList={reviews}
            mostHelpfulFavorable={most_helpful_favorable}
            mostHelpfulCritical={most_helpful_critical}
          />
          <PaginationAndSort
            totalNumberReviews={total_number_reviews}
          />
          {reviews && reviews.map((review) => (
            <IndividualReview
              review={review}
              key={review._id}
            />
          ))}
          <PaginationAndNextPage
            totalNumberReviews={total_number_reviews}
          />
        </div>
      </div>
    );
  }
}

export default App;
