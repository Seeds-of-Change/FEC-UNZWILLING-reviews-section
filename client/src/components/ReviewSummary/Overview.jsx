import React from 'react';
import PropTypes from 'prop-types';
import styles from './ReviewSummary.module.css';

import STAR_IMAGE from '../filled-star.jpg';
import EMPTY_STAR_IMAGE from '../empty-star.png';

import GREY_BAR from '../grey.jpg';
import YELLOW_BAR from '../yellow.png';

const Overview = (props) => {
  Overview.propTypes = {
    numberOfFiveStarReviews: PropTypes.number,
    numberOfFourStarReviews: PropTypes.number,
    numberOfThreeStarReviews: PropTypes.number,
    numberOfTwoStarReviews: PropTypes.number,
    numberOfOneStarReviews: PropTypes.number,
    averageValueRating: PropTypes.number,
    averageQualityRating: PropTypes.number,
    averageStarRating: PropTypes.number,
  };
  Overview.defaultProps = {
    numberOfFiveStarReviews: 0,
    numberOfFourStarReviews: 0,
    numberOfThreeStarReviews: 0,
    numberOfTwoStarReviews: 0,
    numberOfOneStarReviews: 0,
    averageValueRating: 0,
    averageQualityRating: 0,
    averageStarRating: 0,
  };
  const {
    numberOfFiveStarReviews,
    numberOfFourStarReviews,
    numberOfThreeStarReviews,
    numberOfTwoStarReviews,
    numberOfOneStarReviews,
    averageStarRating,
    averageQualityRating,
    averageValueRating,
  } = props;

  function renderStars(num) {
    const stars = Array.apply(null, new Array(5))
    return stars.map((star, index) => (
      <img
        alt="star"
        style={{ width: '15px' }}
        src={index < num ? STAR_IMAGE : EMPTY_STAR_IMAGE}
      />
    ));
  }

  function renderBars(num) {
    const stars = Array.apply(null, new Array(10))
    return stars.map((star, index) => (
      <img
        alt="bar"
        style={{ width: '10px' }}
        src={index < num ? YELLOW_BAR : GREY_BAR}
      />
    ));
  }

  return (
    <div>
      <h4 className={styles.header}>
        Reviews
      </h4>
      <div className={styles.block}>
        <div className={styles.inline_block}>
          <p>Ratings Snapshot</p>

          <p className={styles.select_header}>
            Select a row below to filter reviews.
          </p>
          <div className={styles.ratings_content}>
            <p>
              5 ★&nbsp;
              {renderBars(`${numberOfFiveStarReviews}`)}&nbsp;&nbsp;
              {numberOfFiveStarReviews}
            </p>
            <p>
              4 ★&nbsp;
              {renderBars(`${numberOfFourStarReviews}`)}&nbsp;&nbsp;
              {numberOfFourStarReviews}
            </p>
            <p>
              3 ★&nbsp;
              {renderBars(`${numberOfThreeStarReviews}`)}&nbsp;&nbsp;
              {numberOfThreeStarReviews}
            </p>
            <p>
              2 ★&nbsp;
              {renderBars(`${numberOfTwoStarReviews}`)}&nbsp;&nbsp;
              {numberOfTwoStarReviews}
            </p>
            <p>
              1 ★&nbsp;
              {renderBars(`${numberOfOneStarReviews}`)}&nbsp;&nbsp;
              {numberOfOneStarReviews}
            </p>

          </div>

        </div>

        <div className={styles.inline_block}>
          <p>Average Customer Ratings</p>
          <div className={styles.ratings_content}>
            <p className={styles.stars}>
              {`Overall: still a work in progress
              ${averageStarRating}, rounded  `}
              {renderStars(3)}
            </p>
            <p>
              Quality of Product:
              {averageQualityRating}
            </p>
            <p>
              Value of Product:
              {averageValueRating}
            </p>
          </div>
        </div>
      </div>
    </div>

  );
};

export default Overview;
