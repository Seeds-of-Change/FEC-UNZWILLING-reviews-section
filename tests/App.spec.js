import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import App from '../client/src/components/App';
import Overview from '../client/src/components/ReviewSummary/Overview';

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the Overview Component inside ReviewSummary', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Overview).length).toEqual(1);
  });
});

describe('My Test Suite', () => {
  test('My Test Case', () => {
    expect(true).toEqual(true);
  });
});