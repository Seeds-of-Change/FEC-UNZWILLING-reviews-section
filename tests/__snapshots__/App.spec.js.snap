// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="block"
  >
    <span
      className="hover-hand"
    >
      Reviews  
    </span>
    <span
      className="hover-hand"
    >
      Questions
    </span>
  </div>
  <div>
    <div
      className="block"
    >
      <div
        className="inline-block"
      >
        <h4>
          Reviews
        </h4>
        <p>
          Ratings Snapshot
        </p>
        <p>
          Select a row below to filter reviews.
        </p>
        <p>
          5 ★:
        </p>
        <p>
          4 ★:
        </p>
        <p>
          3 ★:
        </p>
        <p>
          2 ★:
        </p>
        <p>
          1 ★:
        </p>
      </div>
      <div
        className="inline-block"
      >
        <p>
          Average Customer Ratings
        </p>
        <p>
          Overall:
        </p>
        <p>
          Quality of Product:
        </p>
        <p>
          Value of Product:
        </p>
      </div>
    </div>
    <div
      className="block"
    >
      <div
        className="inline-block"
      >
        <p>
          Most Helpful Favorable Review
        </p>
        <p>
          Star Rating:
        </p>
        <span>
          Username:
            
        </span>
        <span>
          Date Posted:
          months ago
        </span>
        <p>
          Review title:
        </p>
        <p>
          Review content:
        </p>
        <p>
          of
          people found this helpful
        </p>
        <p>
          See more 4 and 5 star reviews
        </p>
      </div>
      <div
        className="inline-block"
      >
        <p>
          Most Helpful Critical Review
        </p>
        <p>
          Star Rating:
        </p>
        <span>
          Username:
            
        </span>
        <span>
          Date Posted:
          months ago
        </span>
        <p>
          Review title:
        </p>
        <p>
          Review content:
        </p>
        <p>
          of
          people found this helpful
        </p>
        <p>
          See more 1, 2, and 3 star reviews
        </p>
      </div>
    </div>
    <div
      className="block"
    >
      <span>
         out of 
         reviews
      </span>
    </div>
    <div
      className="block"
    >
      <span>
         out of 
         reviews
      </span>
    </div>
  </div>
</div>
`;
